sound/mpg123/Makefile

```diff
@@ -38,33 +37,54 @@ define Package/libmpg123
   DEPENDS:=+libltdl
 endef
 
+define Package/libout123
+  $(call Package/mpg123/Default)
+  SECTION:=libs
+  CATEGORY:=Libraries
+  TITLE:=Library for continuous playback of audio streams via various platform-specific output methods
+  DEPENDS:=+libltdl
+endef
+
 define Package/mpg123
   $(call Package/mpg123/Default)
   SECTION:=sound
   CATEGORY:=Sound
   TITLE:=fast console mpeg audio player
-  DEPENDS+=+libmpg123 +alsa-lib
+  DEPENDS+=+libmpg123 +alsa-lib +libout123
 endef
 
-define Build/Configure
-	$(call Build/Configure/Default, \
-		--enable-shared \
-		--enable-static \
+CONFIGURE_ARGS+= \
+	--enable-shared \
+	--enable-static \
+	--with-audio=alsa \
+	--with-default-audio=alsa \
+
+ifeq ($(CONFIG_SOFT_FLOAT),y)
+	CONFIGURE_ARGS+= \
 		--with-cpu=generic_nofpu \
-		--with-audio=alsa \
-		--with-default-audio=alsa \
-	)
-endef
+		--enable-int-quality=yes
+else
+	CONFIGURE_ARGS+= \
+		--with-cpu=generic_fpu
+endif
+
+ifneq ($(findstring neon,$(CONFIG_TARGET_OPTIMIZATION)),)
+        CONFIGURE_ARGS+= \
+                --with-cpu=arm_fpu
+endif
 
```

Comment:
I think it's correct to set the pre-processor flags for specifying DEFINE and INCLUDE, It may be a matter of style?